tests/unit/via/services/youtube_transcript_test.py

```diff
@@ -0,0 +1,300 @@
+                    ),
+                ],
+                0,
+            ),
+            (
+                [
+                    TranscriptInfoFactory(
+                        language_code="en", name="English", autogenerated=True
+                    ),
+                    TranscriptInfoFactory(
+                        language_code="en-us",
+                        name="English (United States)",
+                        autogenerated=True,
+                    ),
+                ],
+                0,
+            ),
+            (
+                [
+                    TranscriptInfoFactory(language_code="fr", name="French"),
+                    TranscriptInfoFactory(
+                        language_code="en", name="English", autogenerated=True
+                    ),
+                ],
+                1,
+            ),
+            (
+                [
+                    TranscriptInfoFactory(language_code="fr", name="French"),
+                    TranscriptInfoFactory(language_code="de", name="Deutsch"),
+                ],
+                0,
+            ),
+        ],
+    )
+    def test_pick_default_transcript(
+        self, svc, transcript_infos, expected_default_transcript_index
+    ):
+        assert (
+            svc.pick_default_transcript(transcript_infos)
+            == transcript_infos[expected_default_transcript_index]
+        )
+
+    def test_get_transcript(self, svc, transcript_info, http_service):
+        http_service.get.return_value.text = """
+            <transcript>
+                <text start="0.21" dur="1.387">Hey there guys,</text>
+                <text start="1.597">Lichen&#39; subscribe</text>
+                <text start="4.327" dur="2.063">
+                    &lt;font color=&quot;#A0AAB4&quot;&gt;Buy my merch!&lt;/font&gt;
+                </text>
+            </transcript>
+        """
+
+        transcript = svc.get_transcript(transcript_info)
+        http_service.get.assert_called_once_with(transcript_info.url)
+
+        assert transcript == [
+            {"duration": 1.387, "start": 0.21, "text": "Hey there guys,"},
+            {"duration": 0.0, "start": 1.597, "text": "Lichen' subscribe"},
+            {"duration": 2.063, "start": 4.327, "text": "Buy my merch!"},
+        ]
+
+    def test_get_transcript_error_response(self, svc, transcript_info, http_service):
+        # We get an error response from the YouTube API.
+        http_service.get.side_effect = UnhandledUpstreamException(
+            "Something went wrong"
+        )
+
+        with pytest.raises(UnhandledUpstreamException):
+            svc.get_transcript(transcript_info)
+
+    def test_get_transcript_unexpected_response(
+        self, svc, transcript_info, http_service
+    ):
+        http_service.get.return_value.text = "foo"
+
+        with pytest.raises(ElementTree.ParseError):
+            svc.get_transcript(transcript_info)
```

Comment:
The XML parsing in `get_transcript()` is very accepting and will generally return something even if the XML is quite different from what we expect: it just iterates over all the XML elements and returns the text of each. The one way I could make it crash was if the response body isn't valid XML at all.